controllers/nodehealthcheck_controller_test.go

```diff
@@ -154,68 +143,75 @@ var _ = Describe("Node Health Check CR", func() {
 		})
 	})
 
+	createObjects := func(objects ...client.Object) {
+		for _, obj := range objects {
+			Expect(k8sClient.Create(context.Background(), obj)).To(Succeed())
+		}
+	}
+
+	deleteObjects := func(objects ...client.Object) {
+		for _, obj := range objects {
+			// ignore errors, CRs might be deleted by reconcile
+			_ = k8sClient.Delete(context.Background(), obj)
+		}
+	}
+
 	Context("Reconciliation", func() {
 		var (
-			underTest       *v1alpha1.NodeHealthCheck
-			objects         []runtime.Object
-			reconciler      NodeHealthCheckReconciler
-			upgradeChecker  fakeClusterUpgradeChecker
-			mhcChecker      mhc.DummyChecker
-			reconcileError  error
-			reconcileResult controllerruntime.Result
-			getNHCError     error
+			underTest *v1alpha1.NodeHealthCheck
+			objects   []client.Object
 		)
 
-		var setupObjects = func(unhealthy int, healthy int) {
+		setupObjects := func(unhealthy int, healthy int) {
 			objects = newNodes(unhealthy, healthy, false)
 			underTest = newNodeHealthCheck()
-			remediationTemplate := newRemediationTemplate()
-			objects = append(objects, underTest, remediationTemplate)
+			objects = append(objects, underTest)
 		}
 
 		JustBeforeEach(func() {
-			client := fake.NewClientBuilder().WithRuntimeObjects(objects...).Build()
-			reconciler = NodeHealthCheckReconciler{
-				Client:                      client,
-				Log:                         controllerruntime.Log.WithName("NHC Test Reconciler"),
-				Scheme:                      scheme.Scheme,
-				ClusterUpgradeStatusChecker: &upgradeChecker,
-				MHCChecker:                  mhcChecker,
-				Recorder:                    record.NewFakeRecorder(20),
+			createObjects(objects...)
+			// give the reconciler some time
+			time.Sleep(2 * time.Second)
+			// get updated NHC
+			Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(underTest), underTest)).To(Succeed())
+		})
+
+		AfterEach(func() {
+			// delete all created objects
+			deleteObjects(objects...)
+
+			// delete all remediation CRs
+			cr := newRemediationCR("")
+			crList := &unstructured.UnstructuredList{Object: cr.Object}
+			Expect(k8sClient.List(context.Background(), crList)).To(Succeed())
+			for _, item := range crList.Items {
+				Expect(k8sClient.Delete(context.Background(), &item)).To(Succeed())
 			}
-			reconcileResult, reconcileError = reconciler.Reconcile(
-				context.Background(),
-				controllerruntime.Request{NamespacedName: types.NamespacedName{Name: underTest.Name}})
-			getNHCError = reconciler.Get(
-				context.Background(),
-				ctrlruntimeclient.ObjectKey{Namespace: underTest.Namespace, Name: underTest.Name},
-				underTest)
+
+			// let thing settle a bit
+			time.Sleep(1 * time.Second)
```

Comment:
NIT: I'm Not completely sure why we need this sleep, but in case it's mandatory  I think you can replace with ``time.Sleep(time.Second)``

Reply:
- all tests use the same environment now, with the same reconciler and cluster state. That means resource deletion will trigger reconciles, which need to finish before we go on with the next test
- while technically you are right, I prefer to be explicit about how many seconds we wait here and think it's better readable. I see `time.Second` more like just a unit, like `1s` vs `s` only.